tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp

```diff
@@ -116,6 +116,32 @@ bool RemoveBufferBlockVisitor::updateStorageClass(
     }
   }
 
+  // For pointer-to-struct cases, we may need to update the storage class for
+  // the inner struct fields.
+  if (const auto *innerStructType = dyn_cast<StructType>(innerType)) {
+    auto fields = innerStructType->getFields();
+    std::vector<StructType::FieldInfo> newFields(fields.begin(), fields.end());
```

Comment:
nit: Personally, I prefer
```
llvm::SmallVector<StructType::FieldInfo, 2> newFields;
for (auto field : fields) {
  if (!<condition>) {
    newFields.push_back(field);
    continue;
  }
  ...
  newFields.push_back(newInnerType);
  ...
}
```

Maybe some developer can consider the following code can make dangling pointers.
(I understand this is not the case)
```
std::vector<StructType::FieldInfo> newFields(fields.begin(), fields.end());
...
newFields[i] = newInnerType;
...
```

Reply:
Done